src/main/java/org/truffleruby/language/TruffleBootNodes.java

```diff
@@ -88,52 +89,119 @@ public DynamicObject forceContext() {
         }
     }
 
-    @CoreMethod(names = "main", onSingleton = true, optional = 1)
+    @CoreMethod(names = "main", onSingleton = true)
     public abstract static class MainNode extends CoreMethodArrayArgumentsNode {
 
+        @Child private SnippetNode findSFileSnippetNode;
+        @Child private SnippetNode checkSyntaxSnippetNode;
```

Comment:
It's 10 lines of codes versus 2, how is that better?
I call for consistency here (and again premature optimization).